### PR TITLE
Errors in songscript

### PR DESCRIPTION
**Pull request by IsotoxalDev**
Display errors than outright crash

### CODE REVIEW

File: Editor/Scenes/SongEditor/SongEditor.gd [hunk 1]
```diff
@@ -2,10 +2,13 @@ extends VBoxContainer
 
 # Signals
 signal playback_finished()
+signal song_script_error(error)
 signal track_pressed (name)
 
+onready var BIN = OS.get_executable_path()
 const BASE_SONG_SCRIPT = """extends SongScript
 
+
```
Comment (Technohacker) [inline Editor/Scenes/SongEditor/SongEditor.gd:11]: Extra line? :thinking:

File: Editor/Scenes/SongEditor/SongEditor.gd [hunk 2]
```diff
@@ -34,15 +45,46 @@ func add_track(instrument: Button):
...
+	var error = (err[0] as String).substr(err[0].find("SCRIPT"))\
+		.replace(script_location, "SongScript")\
+		.replace(" (", "")\
+		.replace("SCRIPT ERROR: ", "")\
+		.replace("GDScript::reload", "")
+	var prev = 0
+	for i in error.count("SongScript:"):
+		prev = error.find(")", error.find("SongScript"))
+		error[prev] = ""
+	return error
```
Comment (Technohacker) [inline Editor/Scenes/SongEditor/SongEditor.gd:61]: Is there a more machine-friendly output for this? Or is it possible to use the godot binary as a language server? If not, can we use regex instead to extract these parts?

File: Editor/Scenes/SongEditor/SongEditor.gd [hunk 3]
```diff
@@ -34,15 +45,46 @@ func add_track(instrument: Button):
...
 	if !gui:
 		var file =  File.new()
-		var dir = Directory.new()
-		file.open("res://song.gd", File.WRITE)
-		file.store_string(song_script_editor.text)
+		file.open("user://song.gd", File.WRITE_READ)
+		if in_file != song_script_editor.text:
+			file.store_string(song_script_editor.text)
+#			yield(get_tree(), "idle_frame")
+			in_file = file.get_as_text() # Idk why but the file isn't storing without this Wierd
```
Comment (Technohacker) [inline Editor/Scenes/SongEditor/SongEditor.gd:71]: I'd assume the file has to be closed after you wrote to it :)

**Review by Technohacker — CHANGES_REQUESTED** [review]

File: Editor/Scenes/SongEditor/SongEditor.gd (hunk 2 above)
Comment (IsotoxalDev) [reply Editor/Scenes/SongEditor/SongEditor.gd:61]: There is indeed an LSP but not sure about if the executable for the application would have that... But even if there is a LSP handling all cases....

File: Editor/Scenes/SongEditor/SongEditor.gd (hunk 3 above)
Comment (IsotoxalDev) [reply Editor/Scenes/SongEditor/SongEditor.gd:71]: hmm

File: Editor/Scenes/SongEditor/SongEditor.gd (hunk 1 above)
Comment (IsotoxalDev) [reply Editor/Scenes/SongEditor/SongEditor.gd:11]: Thats based on godot's language formatting. Even if you codebase is not formatted based on gd script formatting rules the inapp code editor must be?

Comment (Technohacker) [reply Editor/Scenes/SongEditor/SongEditor.gd:11]: Ahh then no probs

File: Editor/Scenes/SongEditor/SongEditor.gd (hunk 3 above)
Comment (IsotoxalDev) [reply Editor/Scenes/SongEditor/SongEditor.gd]: Browsing through the docs said i had to flush

**Review by Technohacker — APPROVED** [review]

**Pull request merged**